tests/cloudpickle_test.py

```diff
@@ -2130,6 +2130,53 @@ def test_pickle_importable_typevar(self):
         from typing import AnyStr
         assert AnyStr is pickle_depickle(AnyStr, protocol=self.protocol)
 
+    @unittest.skipIf(sys.version_info < (3, 7),
+                     "Pickling generics not supported below py37")
+    def test_generic(self):
+        from typing import (
+            Optional, TypeVar, Generic, Tuple, Callable,
+            Dict, Any, ClassVar, NoReturn, Union, List,
+        )
+
+        T = TypeVar('T')
+
+        class C(Generic[T]):
+            pass
+
+        objs = [
```

Comment:
New tests in this PR should be exclusively about pickling-depickling classes deriving from `Generic` type annotations. You can remove pretty much every item in `objs` besides `C`.